include/neml2/drivers/ModelDriver.h

```diff
@@ -25,6 +25,7 @@
 #pragma once
 
 #include "neml2/drivers/Driver.h"
+#include "neml2/models/Model.h"
```

Comment:
I think a forward declaration is enough.

Reply:
It is not:

```
TransientDriver.cxx:288:30: error: invalid use of incomplete type 'class neml2::Model'
  288 |   _result_out[_step_count] = _model.value(_in);
```